centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -3305,3 +3306,34 @@ function computeMacroValue(array $macroInformations, int $hostId, Kernel $kernel
 
     return $vaultedMacros['_HOST' . $_REQUEST[$macroOriginalNameKey]] ?? $value;
 }
+
+function validateParentChildAreNotCircular(array $fields): array|true
+{
+    global $pearDB;
+
+    $diff = new BasicDifference($fields['host_parents'] ?? [], $fields['host_childs'] ?? []);
```

Comment:
This is not enought to ensure there is no circular lineage (see case 2 in the ticket's how to reproduce)